src/NexusMods.App.UI/LeftMenu/Items/ApplyControlViewModel.cs

```diff
@@ -40,56 +49,70 @@ public ApplyControlViewModel(LoadoutId loadoutId, IServiceProvider serviceProvid
         _applyService = serviceProvider.GetRequiredService<IApplyService>();
         LaunchButtonViewModel = serviceProvider.GetRequiredService<ILaunchButtonViewModel>();
         LaunchButtonViewModel.LoadoutId = loadoutId;
+        
+        var currentLoadout = _loadoutRegistry.Get(loadoutId);
+        if (currentLoadout is null)
+            throw new ArgumentException("Loadout not found", nameof(loadoutId));
+        
+        _newestLoadout = Observable.Return(currentLoadout)
+            .Merge(_loadoutRegistry.RevisionsAsLoadouts(loadoutId))
+            .OnUI()
+            .ToProperty(this, vm => vm.NewestLoadout);
 
-
-        _gameInstallation = _loadoutRegistry.Get(_loadoutId)?.Installation ??
-                            throw new ArgumentException("Loadout not found: " + _loadoutId);
-
-        (_lastAppliedLoadoutId, _lastAppliedRevisionId) = GetLastAppliedLoadout();
-
-        _newestLoadout = _loadoutRegistry.RevisionsAsLoadouts(loadoutId)
-            .ToProperty(this, vm => vm.NewestLoadout, scheduler: RxApp.MainThreadScheduler);
+        _gameInstallation = currentLoadout.Installation;
         
+        _lastAppliedRevisionId =_applyService.LastAppliedRevisionFor(_gameInstallation)
+            .OnUI()
+            .ToProperty(this, vm => vm.LastAppliedRevisionId);
```

Comment:
Set the scheduler in `ToProperty` to `RxApp.MainThreadScheduler` and remove the `OnUI` call.